smi-conformance/grpc/handlers.go

```diff
@@ -89,17 +95,24 @@ func (s *Service) RunTest(ctx context.Context, req *conformance.Request) (*confo
 			}
 			d.Status = conformance.ResultStatus_FAILED
 			d.Capability = conformance.Capability_NONE
-			failures += 1
-			if (res.Assertions - failures) > (res.Assertions / 2) {
+
+			// A hacky way to see the testStep Failed, since KUDO only provides it in Failure.Message
+			re := regexp.MustCompile(`[0-9]+`)
+			stepFailed := re.FindAllString(res.Failure.Message, 1)
```

Comment:
check if `res.Failure!=nil`

Reply:
Aready checked [here](https://github.com/layer5io/learn-layer5/pull/76/files#diff-baad359c237d9e0cd128eb91bc99f1336175b3ea7d45d7f2658ae49a0d752d2bR83) in `if len(res.Failure.Text) > 2`

Reply:
No, res.Failure nil will result in panic.

Reply:
Okey. Added the panic test.